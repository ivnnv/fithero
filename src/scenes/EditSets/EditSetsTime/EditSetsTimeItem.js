/* @flow */

import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text, useTheme } from 'react-native-paper';

import { formatDuration } from '../../../utils/metrics';

type Props = {|
  time: number,
|};

const EditSetsTimeItem = (props: Props) => {
  const { time } = props;
  const { colors } = useTheme();
  const duration = formatDuration(time);
  const placeholderText = duration.length === 2 ? '0:' : '';
  return (
    <>
      <Text style={styles.text}>
        <Text style={[{ color: colors.disabled }, styles.innerText]}>
          {placeholderText}
        </Text>
        <Text style={styles.innerText}>{duration}</Text>
      </Text>
      <View style={styles.space} />
    </>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 18,
    flex: 0.5,
  },
  innerText: {
    textAlign: 'right',
  },
  space: {
    flex: 0.5,
    paddingLeft: 16,
  },
});

export default EditSetsTimeItem;
