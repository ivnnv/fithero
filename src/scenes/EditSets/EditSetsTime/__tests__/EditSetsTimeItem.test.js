/* @flow */

import React from 'react';
import { render } from 'react-native-testing-library';

import EditSetsTimeItem from '../EditSetsTimeItem';

describe('EditSetsTimeItem', () => {
  it('compares set with placeholder for only seconds', () => {
    const { toJSON: secondsJSON } = render(<EditSetsTimeItem time={45} />);
    const { toJSON: minutesJSON } = render(<EditSetsTimeItem time={110} />);

    // $FlowFixMe
    expect(secondsJSON()).toMatchDiffSnapshot(minutesJSON(), {
      contextLines: 0,
      stablePatchmarks: true,
    });
  });
});
