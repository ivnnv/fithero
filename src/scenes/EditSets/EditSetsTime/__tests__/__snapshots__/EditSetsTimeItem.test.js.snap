// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditSetsTimeItem compares set with placeholder for only seconds 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
-       0:
+       
@@ --- --- @@
-       45
+       1:50
`;
